### PR TITLE
Lesson part navigation

### PR DESCRIPTION
**Pull request by kicferk1**
### Context
We can split lesson parts, and so we can have multiple parts per lesson.
We want to see more than one!

### Changes proposed in this pull request
1. Add navigation list to the top of the lesson part page
2. Add 'previous' and 'next' buttons at the bottom of lesson part page
3. Only show 'complete' option on the final lesson part
4. Sneakily replace all `link_to`s with `govuk_link_to`s. 

### Guidance to review
This is a purely frontend PR.
I made navigation elements into components, cause they look ugly and cause I think we might reuse them.
I stole the navigation look from https://www.gov.uk/universal-credit/eligibility, which in turn stole their css from [govuk publishing components](https://github.com/alphagov/govuk_publishing_components/blob/5d4ed0788b5dd2a9865b45eb7d062eb7f59dd73a/app/assets/stylesheets/govuk_publishing_components/components/_previous-and-next-navigation.scss). 
I renamed `gem-c-pagination` to `step-by-step-navigation`, not because I'm ashamed I stole from publishing components (hehe, govspeak), but because I think that's what we will use those styles for, and readability in our repo trumps consistent class names on the internet in my view.

### Testing
I haven't added any tests. Maybe I should (cypress?). Thought?

### CODE REVIEW

File: app/components/navigation_next_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<li class="step-by-step-navigation__item step-by-step-navigation__item--previous">
+  <%= govuk_link_to(@url, class: "step-by-step-navigation__link") do %>
+    <span class="step-by-step-navigation__link-title">
+      <svg class="step-by-step-navigation__link-icon" xmlns="http://www.w3.org/2000/svg" height="13" width="17" viewBox="0 0 17 13">
```
Comment (seb-siem-softwire) [inline app/components/navigation_next_component.html.erb:4]: Is there a better way of handling swg files - for example, https://stackoverflow.com/questions/36986925/how-do-i-display-svg-image-in-rails/36992492 provides some alternatives - there is a high possibility that we will be using more swg images and there is an opportunity here to create a more elegant solution

File: app/components/navigation_previous_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<li class="step-by-step-navigation__item step-by-step-navigation__item--previous">
+  <%= govuk_link_to(@url, class: "step-by-step-navigation__link") do %>
+    <span class="step-by-step-navigation__link-title">
+      <svg class="step-by-step-navigation__link-icon" xmlns="http://www.w3.org/2000/svg" height="13" width="17" viewBox="0 0 17 13">
+        <path d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"></path>
```
Comment (seb-siem-softwire) [inline app/components/navigation_previous_component.html.erb:5]: Similar to the above - is there a better way of handling svg files

File: app/views/core_induction_programme/lesson_parts/show.html.erb [hunk 3]
```diff
@@ -31,5 +46,21 @@
       <% end %>
     <% end %>
 
+    <nav class="step-by-step-navigation" role="navigation">
+      <ul class="step-by-step-navigation__list">
+        <% if @course_lesson_part.previous_lesson_part %>
+          <%= render NavigationPreviousComponent.new(
```
Comment (seb-siem-softwire) [inline app/views/core_induction_programme/lesson_parts/show.html.erb:52]: Sweet!

File: app/components/navigation_next_component.html.erb (hunk 1 above)
Comment (kicferk1) [reply app/components/navigation_next_component.html.erb:4]: I'll look into this :)

**Review by rupollbabu — APPROVED** [review]
I have tested functionality, working as per Acceptance criteria.

File: app/components/navigation_next_component.html.erb (hunk 1 above)
Comment (callumacrae) [reply app/components/navigation_next_component.html.erb:4]: Or you can use webpack to add it as a background image in an SCSS file

Comment (kicferk1) [reply app/components/navigation_next_component.html.erb]: Opted for the webpack option, cause we already had two images we stole from govuk publishing components there :)

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -28,7 +28,7 @@
     <header class="govuk-header" role="banner" data-module="govuk-header">
       <div class="govuk-header__container govuk-width-container">
         <div class="govuk-header__logo">
-          <%= link_to "/", class: "govuk-header__link govuk-header__link--homepage" do %>
+          <%= govuk_link_to "/", class: "govuk-header__link govuk-header__link--homepage" do %>
             <span class="govuk-header__logotype">
```
Comment (seb-siem-softwire) [inline app/views/layouts/application.html.erb:32]: Could you update this svg one as well

Comment (kicferk1) [reply app/views/layouts/application.html.erb:32]: I'd rather not - it looks complicated (fallback images, what is this?), and more importantly, css for the classes that live on `application` svgs aren't in our code - they are from `govuk-frontend`. So we would have to add our versions of them to our code, with small differences, and then worry about maintaining them.

Also, I am quite happy keeping `application` file complicated and unreadable 😂

Comment (seb-siem-softwire) [reply app/views/layouts/application.html.erb:32]: 👍

**Review by callumacrae — APPROVED** [review]

**Pull request merged**